ComponentKit/Core/CKGroupComponent.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <ComponentKit/CKComponent.h>
+#import <ComponentKit/CKComponentLayout.h>
+
+@interface CKGroupComponent : CKComponent
+
+/*
+ Returns a vector of 'CKComponent' children that will be render to the screen.
+
+ If you overrides this method, you must override the `computeLatoutThatFits:children:` and provide a layout for these components.
+ If you don't need a custom layout, you can just use CKFlexboxComponent instead.
+
+ @param state The current state of the component.
+ */
+- (std::vector<CKComponent *>)renderGroup:(id)state;
+
+/**
+ You have to override this method to computes your component's layout.
+
+ @param constrainedSize A min and max size. This is computed as described in the description. The CKComponentLayout you
+ return MUST have a size between these two sizes. This is enforced by assertion.
+ @param children A vector of children that the component provided in the 'renderGroup:' method.
+ */
+- (CKComponentLayout)computeLayoutThatFits:(CKSizeRange)constrainedSize
+                                  children:(std::vector<CKComponent *>)children;
```

Comment:
@adamjernst, this is a first attempt to address the inconsistency between the components in the `renderGroup:` method and ones we have in the layout. We can assert and make sure that the `CKComponentLayout` contains the same components in its children vector. However, if we assert it it will break the current behaviour of ComponentKit.

Reply:
We probably also need to implement:

```
- (CKComponentLayout)computeLayoutThatFits:(CKSizeRange)constrainedSize
                          restrictedToSize:(const CKComponentSize &)size
                      relativeToParentSize:(CGSize)parentSize
                                  children:(std::vector<CKComponent *>)children
```